lib/ppwm_matcher/app.rb

```
@@ -28,6 +33,21 @@ class App < Sinatra::Base
       def repos
         github_request("user/repos")
       end
+
+      def protected!
+        return if authorized?
+
+        headers['WWW-Authenticate'] = 'Basic realm="Restricted Area"'
+        halt 401, "Not authorized\n"
+      end
+
+      def authorized?
+        @auth ||=  Rack::Auth::Basic::Request.new(request.env)
+        @auth.provided? and
```

Honestly, we could be 'more confident' and just let it blow up if a resource is 'protected!' but credentials weren't passed (or a blanket rescue with a halt)...